feat: add quality tasks - first approximation.

johanseto opened the pull request:
## Description

The PR purpose is to integrate quality tasks into the repo.

## Testing instructions

## Additional information

## Checklist for Merge

- [ ] Add  repo files: license File, readme. changelog
- [x] Update makefile
- [ ] Add setup configuration (setup.py setup.cfg)
- [x] Add lint or quality action.
- [x] Clean python code
- [ ] Rebased master/main
- [ ] Squashed commits

<!--
You can put NA in the checklist if it doesn't apply

- [ X ] Check that don't apply / NA
-->

johanseto committed:
feat: Add license file

Create license file `LICENSE` with the terms of the AGPLv3.

johanseto committed:
clean: fix whitespace in pyfiles.

Fix some whitespace in `codejailservice/*`. Some pep 8 issues.

johanseto committed:
refactor: refactor `Makefile` with more options.

Refactor  file `Makefile` with common options used in other projects
like stackbuilder.

Also change and add common requirements in requirements folder.

johanseto committed:
style: Update py files to pass some quality tasks.

Add `setup.cfg` with pycodestyle,isort config.

Add `pylintrc` for pylint run configuration.

Add the doctring for pyfiles.

In `app.py` change logger to LOG var for good practices.

Add clean option in makefile.

johanseto committed:
feat: github action to run quality tasks.

Add github-action called `test.yml` in
github/workflows/

johanseto committed:
fix: fix configuration for import codejail.

Add configuration in `pylinttrc` and `setup.cfg` to recognize
codejail as a thirdparty lib.

felipemontoya commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,28 @@
+        make quality
+    # - name: Unit tests
+    #   run: |
+    #     make test
```
Mind the EOF with a new line

johanseto replied:
Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

felipemontoya commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,28 @@
+    - name: Run the QA suite
+      run: |
+        make quality
+    # - name: Unit tests
```
Dont leave things commented out. If you want it in, leave it, otherwise remove it.

johanseto replied:
Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

felipemontoya commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,28 @@
+
+on:
+  push:
+    branches: [ main, quality-tasks ]
```
Why not all?

johanseto replied:
Changed to  `'**'` for check all branches in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

felipemontoya commented on LICENSE:
```diff
@@ -0,0 +1,671 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```
Please open a PR only for this

johanseto replied:
I quit the License file in this commit: 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f.
Next: open new PR

felipemontoya commented on Makefile:
```diff
@@ -1,7 +1,57 @@
+	pip-sync requirements/dev.txt
+	#pip install -e .
+
+test: ## run unitary tests and meassure coverage
```
we can leave the unit testing bits for a different PR

johanseto replied:
Quit this section in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

felipemontoya commented on Makefile:
```diff
@@ -1,7 +1,57 @@
+	pycodestyle codejailservice *.py
+	pydocstyle codejailservice *.py
+	isort --check-only --diff --recursive codejailservice *.py
+	#python setup.py bdist_wheel
```
we need to fix this and uncomment

felipemontoya replied:
But also maybe fixing the setup deserves its own PR

johanseto replied:
First: Quit this part in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f.

felipemontoya commented on codejailservice/__init__.py:
```diff
@@ -1 +1,2 @@
+"""Helps you keep your cool when creating dozens of open edX and eduNEXT environments."""
```
This should be a docstring for this module.
Somehting like:



```suggestion
"""Allows for the remote execution of unsafe code via a rest API that runs restricted by an apparmor profile."""
```

felipemontoya commented on codejailservice/app.py:
```diff
@@ -56,22 +62,25 @@ def configure_codejail(app):
 
 @app.route("/")
 def index():
+    """Index route logic."""
```
```suggestion
    """A welcome message to give an idea of what this service entails."""
```

Can we remove the reference to edx? edx->openedx.

"An openedx codejail service developed by edunext."

felipemontoya commented on codejailservice/tutor.py:
```diff
@@ -0,0 +1 @@
+"""Helps you keep your cool when creating dozens of open edX and eduNEXT environments."""
```
do we need this file?

johanseto replied:
In the process, updating the machine to run the codejail service and plugin to resolve it.

felipemontoya commented on pylintrc:
```diff
@@ -0,0 +1,128 @@
+[EXCEPTIONS]
+overgeneral-exceptions = Exception
+
+# 1a67033d4799199101eddf63b8ed0bef3e61bda7
```
EOF

johanv26 replied:
Solved in https://github.com/eduNEXT/codejailservice/pull/3/commits/d881138a9a318bd21bb362a0c0a57c15dee8fdfc.

johanseto replied:
Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

felipemontoya commented on setup.cfg:
```diff
@@ -0,0 +1,16 @@
+exclude=.git,.tvm
+
+[tool:pytest]
+norecursedirs = strains .tvm
```
EOF

johanseto replied:
Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

felipemontoya commented on setup.cfg:
```diff
@@ -0,0 +1,16 @@
+
+[pycodestyle]
+ignore=E501, E722
+exclude=.git,.tvm
```
no need to ignore tvm

johanseto replied:
Not ignored .tvm in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

felipemontoya commented on setup.cfg:
```diff
@@ -0,0 +1,16 @@
+multi_line_output = 3
+known_third_party = codejail
+
+[wheel]
```
This belongs together with the setup.py PR

johanseto replied:
I quit the wheel section in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f.
Next: Make the setup PR

felipemontoya reviewed (changes requested):
Overall a good PR, but we need to split it.
Lets leave this just for QA and open new PRs for Unit testing and setup.py

johanseto committed:
docs: Apply suggestions in .py docstrings

Apply suggestions from code review in docstrings py files.

Co-authored-by: Felipe Montoya <felipe.montoya@edunext.co>

johanseto committed:
fix: solve issues of code review.

Update github action `quality.yml` for check all branches.

Check EOF in `pylintrc`, `setup.cfg`.

Quit test  section for individual PR.

Quit license for individual PR.

Quit wheel building and related with `setup.py` for individual PR.

johanseto commented:
This PR is moved or continue in PR: https://github.com/eduNEXT/codejailservice/pull/6
